stable_baselines3/common/off_policy_algorithm.py

```diff
@@ -102,23 +104,26 @@ def _setup_model(self):
                                         self.lr_schedule, **self.policy_kwargs)
         self.policy = self.policy.to(self.device)
 
-    def save_replay_buffer(self, path: str):
+    def save_replay_buffer(self, path: Union[str, pathlib.Path, io.IOBase]) -> None:
         """
         Save the replay buffer as a pickle file.
 
-        :param path: (str) Path to the file where the replay buffer should be saved
+        :param path: (Union[str,pathlib.Path]) Path to the file where the replay buffer should be saved.
+            if path is a str or pathlib.Path, the path is automatically created if necessary.
         """
         assert self.replay_buffer is not None, "The replay buffer is not defined"
-        with open(path, 'wb') as file_handler:
-            pickle.dump(self.replay_buffer, file_handler)
+        save_to_pkl(path, self.replay_buffer)
 
-    def load_replay_buffer(self, path: str):
+    def load_replay_buffer(self, path: Union[str, pathlib.Path]) -> None:
         """
         Load a replay buffer from a pickle file.
 
-        :param path: (str) Path to the pickled replay buffer.
+        :param path: (Union[str, pathlib.Path]) Path to the pickled replay buffer.
         """
-        with open(path, 'rb') as file_handler:
+        path = pathlib.Path(path)
```

Comment:
I would prefer allowing any suffix here, if the file exists, [similar to how stable-baselines did it](https://github.com/hill-a/stable-baselines/blob/master/stable_baselines/common/base_class.py#L614). If file does not exist, then try with the default .pkl suffix.

Reply:
I see, yes I will add it.